src/aggregator/client/conn.go

```diff
@@ -87,6 +119,9 @@ func newConnection(addr string, opts ConnectionOptions) *connection {
 		maxDuration:    opts.MaxReconnectDuration(),
 		writeRetryOpts: opts.WriteRetryOptions(),
 		rngFn:          rand.New(rand.NewSource(time.Now().UnixNano())).Int63n,
+		compressType:   opts.CompressType(),
+		snappyWriter:   snappy.NewBufferedWriter(nil),
+		flateWriter:    flateWriter,
```

Comment:
Do these all need to be created here? Doesn't seem to be a way to change the compressType, so maybe better to add a `Reset(c *net.TCPConn)` method to `connWriter` interface and just create the appropriate writer here, then call `Reset(tcpConn)` in `connectWithLock()`?